### DISCUSSION TITLE
How to set different manifest configurations for different browsers

### DISCUSSION THREAD

**Discussion by asasas234** [wxt-dev/wxt#442] [Q&A]
I need to add an HTTP URL to the permissions section of the manifest.json for the Firefox browser, but I want to ensure that this addition doesn't affect the output of other manifests. How can I achieve this?

Additionally, I am truly grateful for this framework. I previously mentioned that I planned to rewrite my plugin based on it, and now that I have completely rewritten and successfully published it, the development experience has been fantastic!

**Comment by asasas234** [comment]
I have identified one approach where a manifest can be configured with a function that allows for dynamic determination. However, this would necessitate rewriting the entire previously defined JSON. I would prefer to have a setup where JSON defines the configurations common to all browsers, while a function outlines the personalized settings.

**Comment by aklinker1** [comment]
Yes, for now, you must use a function. 

If you have a proposal for what this may look like, feel free to share it. Here's what I've done in the past with a different tool: https://vite-plugin-web-extension.aklinker1.io/guide/supporting-multiple-browsers.html#manifest-templates
